tests/unit_tests/test_matchmaker_queue_algorithm.py

```diff
@@ -439,3 +445,129 @@ def test_make_matches_communicates_failed_attempts(p):
     assert not matches
     assert s1.failed_matching_attempts == 1
     assert s2.failed_matching_attempts == 1
+
+
+@pytest.mark.parametrize("make_teams_func", (
+    algorithm.make_teams,
+    algorithm.make_teams_from_single
+))
+@given(
+    searches=st_searches_list(max_players=1),
+    size=st.integers(min_value=1, max_value=10)
+)
+def test_make_teams_single_correct_size(searches, size, make_teams_func):
+    matched, non_matched = make_teams_func(searches, size)
+
+    for search in matched:
+        assert len(search.players) == size
+
+
+def test_make_teams_single_2v2_large_pool(p):
+    """
+    When we have a large number of players all with similar ratings, we want
+    teams to be formed by putting players with the same rating on the same team.
+    """
+
+    # Large enough so the test is unlikely to pass by chance
+    num = 40
+
+    searches = [Search([p(1000, 10, name=f"p{i}")]) for i in range(num)]
+    searches += [Search([p(500, 10, name=f"p{i}")]) for i in range(num)]
```

Comment:
Maybe this test could still work and would be a bit more meaningful if the ratings are `1000 + random()*epsilon` for a small epsilon and below we assert that the rating of matched searches differs by at most `2*epsilon`.